ethcore/src/client/client.rs

```diff
@@ -1318,6 +1318,33 @@ impl BlockChainClient for Client {
 		self.do_virtual_call(&env_info, &mut state, &t, analytics)
 	}
 
+*/
+	fn replay(&self, id: TransactionId, analytics: CallAnalytics) -> Result<Executed, CallError> {
+		let address = self.transaction_address(id).ok_or(CallError::TransactionNotFound)?;
+		let block = BlockId::Hash(address.block_hash);
+
+		Ok(self.replay_block_transactions(block, analytics)?.nth(address.index).unwrap())
```

Comment:
Instead of `unwrap` use `expect(<proof>)`. Here the proof will be something like: "The transaction address contains a valid index within block; qed".